addons/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/handler/PowermaxBridgeHandler.java

```diff
@@ -0,0 +1,641 @@
+                currentState.setLastKeepAlive(System.currentTimeMillis());
+            }
+        }
+    }
+
+    public String getInfoSetup() {
+        return (getPanelSettings() == null) ? "" : getPanelSettings().getInfo();
+    }
+
+    @Override
+    public void onNewStateEvent(EventObject event) {
+        PowermaxStateEvent stateEvent = (PowermaxStateEvent) event;
+        PowermaxState updateState = stateEvent.getState();
+
+        if (Boolean.TRUE.equals(currentState.isPowerlinkMode())
+                && Boolean.TRUE.equals(updateState.isDownloadSetupRequired())) {
+            // After Enrolling Powerlink or if a reset is required
+            logger.info("Powermax alarm binding: Reset");
+            commManager.startDownload();
+            if (currentState.getLastKeepAlive() != null) {
+                currentState.setLastKeepAlive(System.currentTimeMillis());
+            }
+        } else if (Boolean.FALSE.equals(currentState.isPowerlinkMode()) && updateState.getLastKeepAlive() != null) {
+            // Were are in standard mode but received a keep alive message
+            // so we switch in PowerLink mode
+            logger.info("Powermax alarm binding: Switching to Powerlink mode");
+            commManager.startDownload();
+        }
+
+        boolean doProcessSettings = (updateState.isPowerlinkMode() != null);
+
+        for (int i = 1; i <= getPanelSettings().getNbZones(); i++) {
+            if (Boolean.TRUE.equals(updateState.isSensorArmed(i))
+                    && Boolean.TRUE.equals(currentState.isSensorBypassed(i))) {
+                updateState.setSensorArmed(i, false);
+            }
+        }
+
+        updateState.keepOnlyDifferencesWith(currentState);
+        updateChannelsFromAlarmState(updateState);
+        currentState.merge(updateState);
+
+        PowermaxPanelSettings panelSettings = getPanelSettings();
+        if (!updateState.getUpdatedZoneNames().isEmpty()) {
+            for (Integer zoneIdx : updateState.getUpdatedZoneNames().keySet()) {
+                if (panelSettings.getZoneSettings(zoneIdx) != null) {
+                    for (PowermaxPanelSettingsListener listener : listeners) {
+                        listener.onZoneSettingsUpdated(zoneIdx, panelSettings);
+                    }
+                }
+            }
+        }
+
+        if (doProcessSettings) {
+            // There is a change of mode (standard or Powerlink)
+            processPanelSettings();
+            commManager.exitDownload();
+        }
+    }
+
+    private void processPanelSettings() {
+        if (commManager.processPanelSettings(currentState.isPowerlinkMode())) {
+            for (PowermaxPanelSettingsListener listener : listeners) {
+                listener.onPanelSettingsUpdated(getPanelSettings());
+            }
+            remainingDownloadAttempts = 0;
+        } else {
+            logger.warn("Powermax alarm binding: setup download failed!");
```

Comment:
Is this a software bug that people should report as an issue? If so, also include the exception here. If not, reduce to debug.

Reply:
It is an important log but I could reduce the level to info.